wap/utils/transform.ts

```diff
@@ -0,0 +1,435 @@
+              name: "imagemOriginal", 
+              url: v.imagemOriginal,
+            }],
+            valueReference: "ATRIBUTO UNICO",
+          },
+        ],
+      })
+    })
+
+
+
+  const selected = hasVariant.find((i) => i.sku == sku) ?? hasVariant[0];
+
+  const isVariantOf: ProductGroup = {
+    "@type": "ProductGroup",
+    hasVariant,
+    productGroupID: product.id.toString(),
+    additionalProperty,
+  };
+
+  return {
+    // @ts-ignore its necessary to product without any variation spec
+    productID: product.id.toString(),
+    name: product.nome,
+    description: product.descricoes.longa,
+    brand: {
+      "@type": "Brand",
+      identifier: product.marca.id.toString(),
+      name: product.marca.nome,
+      logo: product.marca.imagem,
+      url: new URL(product.marca.rota.path, baseUrl).href,
+    },
+    url: `${getUrl(`/${product.rota.params.produto}`, baseUrl).href}.html`,
+    image: images.length ? images : [DEFAULT_IMAGE],
+    category: (product as WapProductDatiled)?.categoria?.nome,
+    isVariantOf,
+    aggregateRating: product.avaliacao.quantidade > 0
+      ? {
+        "@type": "AggregateRating",
+        ratingValue: product.avaliacao.media,
+        ratingCount: product.avaliacao.quantidade,
+        reviewCount: product.avaliacao.avaliacoes?.length,
+      }
+      : undefined,
+    review: product.avaliacao.avaliacoes?.map((avaliacao) => ({
+      "@type": "Review",
+      // "@id": avaliacao.id,
+      author: [{
+        "@type": "Author",
+        name: avaliacao.nome,
+        identifier: avaliacao.email,
+      }],
+      reviewBody: avaliacao?.comentario,
+      datePublished: avaliacao?.data,
+      itemReviewed: String(avaliacao?.produto?.id),
+      reviewRating: {
+        "@type": "AggregateRating",
+        ratingValue: avaliacao.nota,
+        ratingCount: 1,
+        bestRating: 5,
+        worstRating: 1,
+      },
+      // TODO PARSE RATING MEDIA
+      // media: avaliacao.anexos.map((anexo: unknown) => ({})),
+    })),
+
+    offers: toOffers(product, product.status),
+    isRelatedTo: (product as WapProductDatiled).combinacoes?.map((
+      combinacao,
+    ) => toProduct(combinacao.produto, baseUrl)),
+    questions: (product as WapProductDatiled).perguntas?.map((
+      pergunta,
+    ): Question => ({
+      "@type": "Question",
+      answerCount: pergunta.respostas.length,
+      name: pergunta.pergunta,
+      text: pergunta.pergunta,
+      datePublished: pergunta.data,
+      suggestedAnswer: pergunta.respostas.map((resposta): Answer => (
+        {
+          datePublished: resposta.data,
+          text: resposta.resposta,
+          author: [{
+            "@type": "Author",
+            identifier: resposta.email,
+            name: resposta.nome,
+          }],
+        }
+      )),
+      author: [{
+        "@type": "Author",
+        identifier: pergunta.email,
+        name: pergunta.nome,
+      }],
+    })),
+    ...selected,
```

Comment:
I do not understand why ...selected is here, it will replace something you did above. Don't? Do you have an example when it is good?

![image](https://github.com/deco-cx/apps/assets/76620866/e84943d5-715b-4a9e-a669-6de0b02d5e0d)

Reply:
yes, this is the desired behaviour once some products dosent have variations and your infos its on product root, but when they have the info its on her variations